pkg/crc/services/dns/dns.go

```diff
@@ -122,11 +122,21 @@ func CheckCRCPublicDNSReachable(serviceConfig services.ServicePostStartConfig) (
 			curlArgs = append(curlArgs, "--proxy", proxyConfig.HTTPProxy)
 		}
 		if proxyConfig.HTTPSProxy != "" {
+			// if https proxy is set, the last --proxy arg will take precedence over the one previously
+			// set for http proxy
 			curlArgs = append(curlArgs, "--proxy", proxyConfig.HTTPSProxy)
 		}
 		curlArgs = append(curlArgs, "--noproxy", proxyConfig.GetNoProxyString())
 		if proxyConfig.ProxyCAFile != "" {
-			curlArgs = append(curlArgs, "--proxy-cacert", proxyConfig.HTTPProxy)
+			// --proxy-cacert/--cacert replaces the system CAs with the specified one.
+			// If not using MITM proxy, --cacert must *not* be used, and if not using
+			// https:// proxy, --proxy-cacert must *not* be used
+			// ProxyCAFile is ambiguous, we cannot know if it's set because of MITM proxy,
+			// because of https:// proxy, or because of both
+			// We do not really care about transport security for this test, all that
+			// matters is whether or not we can resolve the hostname, so we can
+			// workaround this ambiguity by using an insecure connection
+			curlArgs = append(curlArgs, "--insecure", "--proxy-insecure")
```

Comment:
@cfergeau with these changes, I still get the same error:

```
WARN Failed public DNS query from the cluster: ssh command error:
command : curl --head quay.io --proxy http://localhost:8080 --proxy http://localhost:8080 --noproxy 127.0.0.1,localhost,.testing --insecure --proxy-insecure
err     : Process exited with status 7
 :  
```

1. There are still two `--proxy` options in the command. Not sure why, maybe one of them can be removed with as part of this PR? (this does not cause the error though)
2. When I run the command manually, I see it go through the proxy and the exit code is not 7, but 0. 
```
$ curl --head quay.io --proxy http://localhost:8080 --proxy http://localhost:8080 --noproxy 127.0.0.1,localhost,.testing --insecure --proxy-insecure
HTTP/1.1 301 Moved Permanently
Connection: keep-alive
Content-Length: 134
Content-Type: text/html
Date: Tue, 12 Apr 2022 20:18:45 GMT
Location: https://quay.io:443/
Server: awselb/2.0

$ echo $?
0
```
And on the proxy I see this:
```
2022/04/12 22:18:45 [026] INFO: Got request / quay.io HEAD http://quay.io/
2022/04/12 22:18:45 [026] INFO: Sending request HEAD http://quay.io/
2022/04/12 22:18:45 [026] INFO: Received response 301 Moved Permanently
2022/04/12 22:18:45 [026] INFO: Copying response to client 301 Moved Permanently [301]
2022/04/12 22:18:45 [026] INFO: Copied 0 bytes to client error=<nil>
```
So it seems that something else is going on here... When you run your PR, do you *not* get the warning?